From 6c4824b9ee3272c7621639f873b6a4b38b5e117e Mon Sep 17 00:00:00 2001
From: Takayuki 'January June' Suwa <jjsuwa_sys3175@yahoo.co.jp>
Date: Thu, 28 Jul 2022 11:59:00 +0900
Subject: [PATCH] xtensa: Add RTX costs for if_then_else

It takes one machine instruction for both conditional branch and move.

gcc/ChangeLog:

	* config/xtensa/xtensa.c (xtensa_rtx_costs):
	Add new case for IF_THEN_ELSE.
---
 gcc/config/xtensa/xtensa.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gcc/config/xtensa/xtensa.c b/gcc/config/xtensa/xtensa.c
index bd3489bfe..b6f41a478 100644
--- a/gcc/config/xtensa/xtensa.c
+++ b/gcc/config/xtensa/xtensa.c
@@ -4309,6 +4309,7 @@ xtensa_rtx_costs (rtx x, machine_mode mode, int outer_code,
 
     case ZERO_EXTRACT:
     case ZERO_EXTEND:
+    case IF_THEN_ELSE:
       *total = COSTS_N_INSNS (1);
       return true;
 
-- 
2.20.1

